flang/runtime/command.cpp

```diff
@@ -282,4 +282,40 @@ std::int32_t RTNAME(GetEnvVariable)(const Descriptor &name,
   return StatOk;
 }
 
+const char *ensureNullTerminated(
+    const char *str, size_t length, Terminator &terminator) {
+  if (length <= strlen(str)) {
+    char *newCmd{(char *)malloc(length + 1)};
+    if (newCmd == NULL) {
+      terminator.Crash("Command not null-terminated, memory allocation failed "
+                       "for null-terminated newCmd.");
+    }
+
+    strncpy(newCmd, str, length);
+    newCmd[length] = '\0';
+    return newCmd;
+  } else {
+    return str;
+  }
+}
+
+void RTNAME(System)(const Descriptor &command, const Descriptor *exitstat,
+    const char *sourceFile, int line) {
+  Terminator terminator{sourceFile, line};
+
+  const char *newCmd{ensureNullTerminated(
+      command.OffsetElement(), command.ElementBytes(), terminator)};
+  int status{std::system(newCmd)};
+
```

Comment:
This blank line should instead deallocate any storage allocated by `EnsureNullTerminated`.  Otherwise it's just going to leak.  Please don't leak memory.

Reply:
Thanks for pointing this out, I've used `FreeMemory()` from `memory.h`. Similar problem that exists on `execute_command_line` has been addressed. https://github.com/llvm/llvm-project/pull/74077#issuecomment-1856249162